classes/village/VillageUpgradeManager.php

```diff
@@ -156,6 +156,14 @@ public static function getBuildingUpgradesForDisplay(System $system, Village $vi
                 foreach (VillageBuildingConfig::UPGRADE_SET_UPGRADES[$upgrade_set_key] as $upgrade_key) {
                     $upgrade_config_data = VillageUpgradeManager::$UPGRADE_CONFIGS[$upgrade_key];
                     $upgrade = $village->upgrades[$upgrade_key] ?? null;
+                    if (isset($upgrade)) {
+                        if ($upgrade->research_progress != null) {
+                            $research_time_remaining = ($upgrade->research_progress_required - $upgrade->research_progress) / $village->research_speed;
+                            $research_time_remaining = System::TimeRemaining($research_time_remaining, format: "long", include_seconds: false, include_minutes: false);
```

Comment:
Hmm, I wonder if there's a case for us to use the raw time value. Could consider storing the raw value in the DTO and then format it to a string where needed (can be a helper method on the `VillageUpgradeDTO` / `VillageUpgrade` class)